backend/WebAPI/Controllers/DesksController.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using backend.Data;
+using backend.Dtos;
+using backend.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using WebAPI.Data;
+
+namespace backend.Controllers
+{
+    [Route("api/[controller]")]
+    
+    public class DesksController : ControllerBase
+    {
+        private const string CollectionName = "DesksCollection";
+       private readonly ICosmosDbRepository<Desk> _repository;
+        public DesksController(ICosmosDbRepository<Desk> repository)
+        {
+            _repository = repository;
+        }
+        [HttpPost("register")]
+         public async Task<IActionResult> RegisterDesk([FromBody] Desk deskForRegister)
+        {
+            await _repository.InsertEntityAsync(CollectionName, deskForRegister);
+        
+            return StatusCode(201);
+        }
+
+
+
+        // PUT api/Desks/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put(string id, [FromBody] Desk deskDto)
+        {
+            deskDto.Id = Guid.Parse(id).ToString();
```

Comment:
Guid Parse + ToString da ci w rezultacie to samo, jedyne co to wysypie się jeśli nie podasz guida ale to lepiej TryParse zrobić